tests/unit/src/test/scala/tests/inlayHints/InlayHintsHoverSuite.scala

```diff
@@ -0,0 +1,217 @@
+       |
+       |```scala
+       |final abstract class Int: Int
+       |```
+       |`Int`, a 32-bit signed integer (equivalent to Java's `int` primitive type) is a
+       | subtype of [scala.AnyVal](scala.AnyVal). Instances of `Int` are not
+       | represented by an object in the underlying runtime system.
+       |
+       | There is an implicit conversion from [scala.Int](scala.Int) => [scala.runtime.RichInt](scala.runtime.RichInt)
+       | which provides useful non-primitive operations.
+       |""".stripMargin,
+  )
+
+  check(
+    "implicit-conversion",
+    """|case class User(name: String)
+       |object Main {
+       |  implicit def intToUser(x: Int): User = new User(x.toString)
+       |  val y: User = <<intToUser>>1
+       |}
+       |""".stripMargin,
+    """|```scala
+       |implicit def intToUser(x: Int): User
+       |```
+       |""".stripMargin,
+  )
+
+  check(
+    "import-rename",
+    """|import scala.collection.{AbstractMap => AB}
+       |import scala.collection.{Set => S}
+       |
+       |object Main {
+       |  def test(d: S[Int], f: S[Char]): AB[Int, String] = {
+       |    val x = d.map(_.toString)
+       |    val y = f
+       |    ???
+       |  }
+       |  val x<<: AB[Int,String]>> = test(Set(1), Set('a'))
+       |}
+       |""".stripMargin,
+    """|```scala
+       |abstract class AbstractMap[K, V]: AbstractMap
+       |```
+       |Explicit instantiation of the `Map` trait to reduce class file size in subclasses.
+       |
+       |```scala
+       |final abstract class Int: Int
+       |```
+       |`Int`, a 32-bit signed integer (equivalent to Java's `int` primitive type) is a
+       | subtype of [scala.AnyVal](scala.AnyVal). Instances of `Int` are not
+       | represented by an object in the underlying runtime system.
+       |
+       | There is an implicit conversion from [scala.Int](scala.Int) => [scala.runtime.RichInt](scala.runtime.RichInt)
+       | which provides useful non-primitive operations.
+       |
+       |```scala
+       |type String: String
+       |```
+       |The `String` type in Scala has all the methods of the underlying
+       | [java.lang.String](java.lang.String), of which it is just an alias.
+       |
+       | In addition, extension methods in [scala.collection.StringOps](scala.collection.StringOps)
+       | are added implicitly through the conversion [augmentString](augmentString).
+       |""".stripMargin,
+  )
+
+  check(
+    "import-rename2",
+    """|object Main {
+       |  import scala.collection.{Set => S}
+       |  import Foo._
+       |
+       |  def test(d: S[Int])(implicit foo: S[Int]): Int = {
+       |    val x = d.map(_.toString)
+       |    ???
+       |  }
+       |
+       |  val x = test(Set(1))<<(ttt)>>
+       |}
+       |
+       |object Foo {
+       |  implicit val ttt: Set[Int] = Set(1)
+       |}
+       |""".stripMargin,
+    // TODO: We should show info about renames (`val ttt: S[Int]` instead of `val ttt: Set[Int]`)
```

Comment:
That should be shown properly with the recent changes by @kasiaMarek right?

Reply:
It's not possible to make it work now, because we the tooltip we show, comes from `hover` on a `ttt` definition. On `ttt` definition there is no information about a rename from `Set` to `S`.

But I think thats not a big issue

Reply:
Ok sure, we can leave it for now, not super important. I guess we would need to handle the resolve request in the compiler to make that work. And it's maybe not worth it.

Reply:
Especially that the user will see the rename + what it dealiases to, so maybe that is better.

Reply:
Yes, and if there are different renames in hint position and symbol definition position, the logic could get quite messy

Reply:
I feel like this could be handled if we were passing two positions to hover, one of the symbol we want to hover on, and the other one where the hover info will actually be shown, so the context for the second one could be used for printing. But it may be too optimistic to assume it will just work, and definitely not worth to consider in this PR. I still think we should create an issue considering this, we can give `low-priority` and leave it be, but it will be there for sake of documentation.